src/main/java/com/commercetools/sync/commons/exceptions/InvalidProductTypeException.java

```diff
@@ -0,0 +1,40 @@
+package com.commercetools.sync.commons.exceptions;
+
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class InvalidProductTypeException extends Exception {
+    private Set<Throwable> causes;
+
+    public InvalidProductTypeException(@Nonnull final String message, @Nonnull final Set<Throwable> causes) {
+        super(buildMessage(message, causes));
+        this.causes = causes;
+    }
+
+    @Nonnull
+    private static String buildMessage(@Nonnull final String message, @Nonnull final Set<Throwable> causes) {
+        final String causesString = asMessage(causes);
+        return isNotBlank(causesString) ? format("%s Causes:%n\t\t%s", message, causesString) :  message;
+    }
+
+    @Nonnull
+    private static String asMessage(@Nonnull final Set<Throwable> causes) {
+        final List<String> causesMessages = causes
+            .stream()
+            .map(Throwable::getMessage)
+            .filter(StringUtils::isNotBlank)
+            .collect(Collectors.toList());
+        return String.join("\n\t\t", causesMessages);
```

Comment:
should it be "\n\t" ?

Reply:
no here the point is to have an extra indentation so in the end it looks like that:

```java
com.commercetools.sync.commons.exceptions.ReferenceReplacementException: Some errors occurred during reference replacement. Causes:
	Failed to replace some references on the productType with key 'withNestedTypeAttr'. Causes:
		Failed to replace some references on the attributeDefinition with name 'nestedattr'. Cause: ProductType reference is not expanded.
		Failed to replace some references on the attributeDefinition with name 'nestedattr2'. Cause: ProductType reference is not expanded.
```